src/utils/filterLinks.ts

```diff
@@ -0,0 +1,18 @@
+import { Link } from '@/types/Folder.types';
+
+function filterLinks(links?: Link[], keyword?: string) {
+  if (!links) return;
+  if (!keyword) return links;
+
+  const lowered = keyword.toLowerCase().split(' ').join('');
+  return links?.filter(({ url, title, description }) => {
+    const searchText = `${url}${title}${description}`
+      .toLowerCase()
+      .split(' ')
+      .join('');
```

Comment:
공백제거의 목적이라면 정규식을 사용하시는 것이 가독성과 성능 측면에서 유리합니다
(`\s`는 공백, `+`는 하나 이상, `g`는 전역 검색을 의미합니다.)

```suggestion
     .replace(/\s+/g, '');
```